RiotSwiftUI/Modules/Settings/Notifications/Service/MatrixSDK/MXNotificationSettingsService.swift

```diff
@@ -71,16 +70,54 @@ class MXNotificationSettingsService: NotificationSettingsServiceType {
         session.notificationCenter.removeRule(rule)
     }
     
-    func updatePushRuleActions(for ruleId: String, enabled: Bool, actions: NotificationActions?) {
-        guard let rule = session.notificationCenter.rule(byId: ruleId) else { return }
-        session.notificationCenter.enableRule(rule, isEnabled: enabled)
+    func updatePushRuleActions(for ruleId: String,
+                               enabled: Bool,
+                               actions: NotificationActions?,
+                               completion: ((Result<Void, Error>) -> Void)?) {
         
-        if let actions = actions {
-            session.notificationCenter.updatePushRuleActions(ruleId,
-                                                             kind: rule.kind,
-                                                             notify: actions.notify,
-                                                             soundName: actions.sound,
-                                                             highlight: actions.highlight)
+        guard let rule = session.notificationCenter.rule(byId: ruleId) else {
+            completion?(.success)
```

Comment:
Shouln't this be failure if we cant find given rule?

Reply:
The rationale is: 
If a rule isn't in the list it doesn't exist. So we don't need to update it.
This is useful to handle gracefully unstable rules turning into stable ones.

Reply:
Makes sense, thanks